Problem037/C++/solution_1.cpp

```diff
@@ -0,0 +1,49 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+const int much = 1000000;
+bool is_prime[much];
+
+void sieve(){
+	memset (is_prime, true, sizeof(is_prime));
+	is_prime[0] = is_prime [1] = false;
+	for (int i = 2; i < much; ++i) {
+		for (int j = (i << 1); j < much; j += i) {
+			is_prime[j] = false;
+		}
+	}
+}
+
+bool test(int t)
+{
+	int orig = t, cnt = 0, tmp10 = 1;
+	while (t > 0) {
+		if (!is_prime [t]) {
```

Comment:
Remove the space character before the `[...]`.

```suggestion
		if (!is_prime[t]) {
```